posthog/src/main/java/com/posthog/internal/PostHogQueue.kt

```diff
@@ -163,32 +174,33 @@ internal class PostHogQueue(
 
         val events = mutableListOf<PostHogEvent>()
         for (file in files) {
-            try {
-                val inputStream = config.encryption?.decrypt(file.inputStream()) ?: file.inputStream()
-                inputStream.use {
-                    val event = config.serializer.deserialize<PostHogEvent?>(it.reader().buffered())
-                    event?.let { theEvent ->
-                        events.add(theEvent)
+            if (file.isStreamable()) {
+                try {
+                    events.add(file.event(config))
+                } catch (e: Throwable) {
+                    synchronized(dequeLock) {
+                        deque.remove(file)
                     }
+                    file.deleteSafely(config)
+                    config.logger.log("File: ${file.name()} failed to parse: $e.")
                 }
-            } catch (e: Throwable) {
-                synchronized(dequeLock) {
-                    deque.remove(file)
-                }
-                file.deleteSafely(config)
-                config.logger.log("File: ${file.name} failed to parse: $e.")
+            } else {
+                events.add(file.event(config))
             }
         }
 
         var deleteFiles = true
         try {
+            // TODO: python checks if the payload isnt bigger than 900kb
             if (events.isNotEmpty()) {
                 when (endpoint) {
                     PostHogApiEndpoint.BATCH -> api.batch(events)
                     PostHogApiEndpoint.SNAPSHOT -> api.snapshot(events)
                 }
             }
         } catch (e: PostHogApiError) {
+            // TODO: python retries up to 3 times with a 500ms interval
+            // if the API returns between 400 and 500
```

Comment:
This probably has to be implemented in case we think its important to retry up to a few times even if the API returns an error.
Right now we drop the events since we prefer to not be shooting the API every time in case there's an issue with the backend.